services/web/server/src/simcore_service_webserver/projects/projects_api.py

```diff
@@ -184,19 +188,94 @@ async def remove_services_and_data():
 ## PROJECT NODES -----------------------------------------------------
 
 
-@observe(event="SIGNAL_PROJECT_CLOSE")
+async def retrieve_and_notify_project_locked_state(
+    user_id: int, project_uuid: str, app: web.Application
+):
+    project = await get_project_for_user(app, project_uuid, user_id, include_state=True)
+    await notify_project_state_update(app, project)
+
+
+# TODO: Once python 3.8 is in use this
```

Comment:
```suggestion
import sys
if sys.version_info >= (3, 8):
    raise Exception("use below implementation in py3.8+")
```

this way you won't miss it

Reply:
that is a good point ... 
I enabled it in py3.8 branch